src/components/PreferencesPage.vue

```diff
@@ -185,63 +185,84 @@
     </label>
     <label class="pref" for="chkSkipSponsors">
         <strong v-t="'actions.skip_sponsors'" />
-        <input id="chkSkipSponsors" v-model="skipSponsor" class="checkbox" type="checkbox" @change="onChange($event)" />
+        <select id="chkSkipSponsors" v-model="skipSponsor" @change="onChange($event)">
+            <option value="off">Disable</option>
+            <option value="skip">Auto Skip</option>
+            <option value="manual">Manual Skip</option>
+            <option value="show">Show</option>
+        </select>
```

Comment:
Rather than duplicating this code, please turn it into a component!

Reply:
I'm not familiar with Vue, do you have any tips on how to do this?  I tried moving the select element into a component but it broke the preference value being saved/loaded.

Reply:
You might want to have a look at the `WatchOnYouTubeButton` which is used as a component, in the end components just work like normal Vue pages/files!

Reply:
I think I have it kind of figured out, but I still can't figure out how to do the onChange portion.

Reply:
Ideally, I would move that portion of the code inside the component too, I could help with that by pushing into this branch too

Reply:
I'd be thankful for any help getting it to work, this is pretty much my first time working with Vue so I don't really know what I'm doing.